src/main/java/org/grpc/service/ShiftSwitchReplyGrpcImpl.java

```diff
@@ -0,0 +1,197 @@
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+            return;
+        }
+
+        replyToUpdate.setDetails(request.getDetails());
+        ShiftSwitchReply replyToSendBack = replyRepository.save(replyToUpdate);
+
+        responseObserver.onNext(dtoConverter.convertReplyToReplyDTO(replyToSendBack));
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getSingleById(Id request, StreamObserver<ReplyDTO> responseObserver) {
+        ShiftSwitchReply requestedReply = replyRepository.findById(request.getId());
+        if (requestedReply == null){
+            Status status = Status.newBuilder()
+                    .setCode(Code.NOT_FOUND_VALUE)
+                    .setMessage("A reply with this ID not found")
+                    .build();
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+            return;
+        }
+
+        ReplyDTO replyDTO = dtoConverter.convertReplyToReplyDTO(requestedReply);
+        responseObserver.onNext(replyDTO);
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getAll(Empty request, StreamObserver<ReplyDTOList> responseObserver) {
+        ArrayList<ShiftSwitchReply> allReplies = replyRepository.findAll();
+        ArrayList<ReplyDTO> replyDTOS = new ArrayList<>();
+        allReplies.forEach(reply -> replyDTOS.add(dtoConverter.convertReplyToReplyDTO(reply)));
+        ReplyDTOList replyDTOList = ReplyDTOList.newBuilder().addAllDtos(replyDTOS).build();
+        responseObserver.onNext(replyDTOList);
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getAllRepliesByEmployeeId(Id request, StreamObserver<ReplyDTOList> responseObserver) {
+        Employee employee = employeeRepository.findById(request.getId());
+        if (employee == null){
+            Status status = Status.newBuilder()
+                    .setCode(Code.NOT_FOUND_VALUE)
+                    .setMessage("A reply with this ID not found")
+                    .build();
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+            return;
+        }
+
+//        List<ShiftSwitchReply> repliesByEmployee = replyRepository.findAll().stream().filter(reply -> reply.getTargetEmployee().getShifts().contains(employee)).toList();
+//        This ^^^ (if it worked) can be substituted by one simple call to the Repository vvv :> [TODO: DELETE THE COMMENT AFTER VIEWING IT]
+        List<ShiftSwitchReply> repliesByEmployee = replyRepository.findAllByTargetEmployeeId(request.getId());
+        List<ReplyDTO> replyDTOS = repliesByEmployee.stream().map(dtoConverter::convertReplyToReplyDTO).toList();
+        responseObserver.onNext(ReplyDTOList.newBuilder().addAllDtos(replyDTOS).build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void setAcceptReplyOrigin(IdBooleanPair request, StreamObserver<Boolean> responseObserver) {
```

Comment:
whats response observer

Reply:
responseObserver is a class that handles feedback to the client. Think about it as the return type of the method, you're calling it onCompleted() if the method is done, onNext() if it has to pass some arguments and onError() if it terminated with an error